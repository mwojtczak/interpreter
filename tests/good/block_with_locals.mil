function main () {
	x = 5;
	a = 1;
	b = 2;
	c = 3;
	print(a);
	print(b);
	print(c);
	[local x];
	[local a];
	[local b];
	[local c];
	{
		a = 10;
		b = 20;
		c = 30;
		print(a);
		print(b);
		print(c);
		[local a];
		[local b];
		[local c];
		{
			a = 100;
			b = 200;
			c = 300;
			print(a);
			print(b);
			print(c);
//block without local decl			
			{
				a = 9;
				b = 99;
				c = 999;
			}
			print(a);
			print(b);
			print(c);
		}
		print(a);
		print(b);
		print(c);
	}
	print(a);
	print(b);
	print(c);
}