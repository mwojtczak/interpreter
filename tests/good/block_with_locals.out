1
2
3
10
20
30
100
200
300
9
99
999
10
20
30
1
2
3
